internal/checker/flow.go

```diff
@@ -1288,10 +1290,10 @@ func (c *Checker) getUnionOrEvolvingArrayType(f *FlowState, types []*Type, subty
 }
 
 func (c *Checker) getTypeAtFlowLoopLabel(f *FlowState, flow *ast.FlowNode) FlowType {
-	if f.refKey == "" {
-		f.refKey = c.getFlowReferenceKey(f)
+	if !f.refKeyValid {
+		f.refKey, f.refKeyValid = c.getFlowReferenceKey(f)
```

Comment:
Can we just use the zero value here to indicate an invalid key? I mean, that's no more likely to collide than a regular valid key.

Reply:
Yeah, I was flip flopping between them.